source/mongodb-handshake/handshake.rst

```diff
@@ -224,6 +224,49 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
+Supporting Wrapping Libraries
+=============================
+
+Drivers MUST allow libraries which wrap the driver to append to the client
+metadata generated by the driver. The following class definition defines the
+options which MUST be supported:
+
+.. code:: typescript
+
+  class DriverInfoOptions {
+    /**
+     * The name of the library wrapping the driver.
+     */
+    name: Optional<String>;
+
+    /**
+     * The version of the library wrapping the driver.
+     */
+    version: Optional<String>;
+
+    /**
+     * Optional platform information for the wrapping driver.
+     */
+    platform: Optional<String>;
+  }
+
+Note that how these options are provided to a driver is left up to the implementor.
+
+If provided, these options MUST not replace the values used for metadata generation, but
+instead append to the respective fields, delimited by a `|` character. For example, when
+`Motor <https://docs.mongodb.com/ecosystem/drivers/motor/>`_ wraps PyMongo, the following
+fields are updated to include Motors "driver info":
```

Comment:
To clarify, would this entail changing existing behavior in any driver that already allows handshake fields to be appended and uses a different delimiter? For instance, libmongoc's `mongoc_handshake_data_append` uses ` / ` (with surrounding spaces).

I assume the MUST at the start of this sentence (prohibiting replacing existing data) still applies to this, but it may be clear if we split the sentence in two. For example:

> Additional options MUST be appended to the respective fields and be delimited by a `|` character.

Also, it may be prudent to discuss priorities should part of the handshake need to be truncated for space limitations. This probably belongs in its own section, but it seems relevant to this PR since we're introducing new API to append more data.

To use libmongoc's existing [`mongoc_handshake_append()`](https://github.com/mongodb/mongo-c-driver/blob/1.15.1/src/libmongoc/src/mongoc/mongoc-handshake.c#L609) function as an example, it allows appending to the driver name, version, and platform fields. The only truncation is applied to the "platform" field in [`_mongoc_handshake_build_doc_with_application()`](https://github.com/mongodb/mongo-c-driver/blob/1.15.1/src/libmongoc/src/mongoc/mongoc-handshake.c#L509), and we don't allow other fields be truncated. I assume the logic there is that platform is a nice-to-have field and is most likely to overflow (libmongoc tends to stuff CFLAGS in there), but if we had to truncate the driver name or other essential fields the handshake data itself wouldn't be too useful so we just fail completely rather than truncate.

Reply:
I updated the section on truncating below, and split the sentence here. What do you propose we do about existing deviations in choice of delimiter?

Reply:
I added a section called "Deviations" where I address this and other deviations brought up here, PTAL. /cc @jyemin